### ISSUE TITLE
Training Anime like ARCANE (Netflix)

### ISSUE THREAD

**Issue by enzyme69** [bryandlee/animegan2-pytorch#17]
Quick question:
Is it possible to train with new style like from Netflix style animation "ARCANE" I really love their rendering of face.

If possible, is it hard, does it take a long time using M1?

**Comment by Greg8978** [comment]
It would be cool indeed. I think Face Portrait v2 is close to this.
I will try to train with my own dataset and will let you know if I succeed with ARCANE pictures

**Comment by zhanglonghao1992** [comment]
@Greg8978 Have you trained the model on your own dataset and what kind of dataset is it, facial animation? Looking forward for your update.

**Comment by GregoireMilliez** [comment]
Hey there!

I collect some images directly from the show.
I try to train the model but it took 187 hours on my ubuntu.
I did not take the time yet to make the setup on my windows computer to use GPU for trainning.

**Comment by zhanglonghao1992** [comment]
> Hey there!
> 
> I collect some images directly from the show. I try to train the model but it took 187 hours on my ubuntu. I did not take the time yet to make the setup on my windows computer to use GPU for trainning.

Would you mind sharing the dataset? I have enough GPU resources for training.

**Comment by Greg8978** [comment]
Sure, have a look [here](https://we.tl/t-1wPGKRJlUJ)

**Comment by Greg8978** [comment]
@zhanglonghao1992, if you can share the training output it would make things easier for me ;)

**Comment by zhanglonghao1992** [comment]
@Greg8978  I failed to train face stylization on the data set you provided. I guess the training style data should contain a large number of clear faces. At present, I plan to collect more face images in Arcane for training.

**Comment by Greg8978** [comment]
Ha ok, thanks for the feedback.

**Comment by AK391** [comment]
this dataset might be of interest, not exactly arcane look but artstation has a realistic stylized look https://github.com/onion-liu/aahq-dataset

**Comment by bryandlee** [comment]
I'm training one but the results are not as good so far. will let you know if I get it work. 
In the mean time, here are some super-cherry-picked golden samples:

![arcane](https://user-images.githubusercontent.com/26464535/144731793-51f73a57-8f36-4243-b515-c83c4db5776c.jpeg)

**Comment by bilal2vec** [comment]
any chance that you could release the current checkpoint? the results look amazing already!

**Comment by zhanglonghao1992** [comment]
> I'm training one but the results are not as good so far. will let you know if I get it work. In the mean time, here are some super-cherry-picked golden samples:
> 
> ![arcane](https://user-images.githubusercontent.com/26464535/144731793-51f73a57-8f36-4243-b515-c83c4db5776c.jpeg)

Did you do face alignment when training face stylization?

**Comment by ymzlygw** [comment]
> Face Por

Thanks, your results looks fine enough. Could you please tell me which training code you use? I have another custom dataset and want to train it too.

**Comment by Sxela** [comment]
I've also given a try to this style. So far I've got results like this on images in the wild. 
![image](https://user-images.githubusercontent.com/11751592/144801328-546d2c6c-8abf-4541-941c-ffaae959eb49.png)

**Comment by zhanglonghao1992** [comment]
> I've also given a try to this style. So far I've got results like this on images in the wild. ![image](https://user-images.githubusercontent.com/11751592/144801328-546d2c6c-8abf-4541-941c-ffaae959eb49.png)

That's cool! Would you mind sharing you training datasets and strategies?

**Comment by Sxela** [comment]
> I'm training one but the results are not as good so far. will let you know if I get it work. In the mean time, here are some super-cherry-picked golden samples:
> 
> ![arcane](https://user-images.githubusercontent.com/26464535/144731793-51f73a57-8f36-4243-b515-c83c4db5776c.jpeg)

Those are awesome! Is that a blend with the same Z or a projection?

**Comment by rainsoulsrx** [comment]
> I'm training one but the results are not as good so far. will let you know if I get it work. In the mean time, here are some super-cherry-picked golden samples:
> 
> ![arcane](https://user-images.githubusercontent.com/26464535/144731793-51f73a57-8f36-4243-b515-c83c4db5776c.jpeg)

Hi, how many pictures did you use to train stylegan model?

**Comment by tinapan-pt** [comment]
> 

Amazing work! When I train stylegan model using screenshots in anime, there are always artifacts on the face. I am troubled by the lack of appropriate data sets. I wonder how many pictures did you use to train stylegan model? I would appreciate it if you could let me know. And if it is convenient, I would like to ask if you can release the data set?

**Comment by tinapan-pt** [comment]
> I'm training one but the results are not as good so far. will let you know if I get it work. In the mean time, here are some super-cherry-picked golden samples:
> 
> ![arcane](https://user-images.githubusercontent.com/26464535/144731793-51f73a57-8f36-4243-b515-c83c4db5776c.jpeg)

Amazing work! When I train stylegan model using screenshots in anime, there are always artifacts on the face. I am troubled by the lack of appropriate data sets. I wonder how many pictures did you use to train stylegan model? I would appreciate it if you could let me know. And if it is convenient, I would like to ask if you can release the data set?

**Comment by chenhk-chn** [comment]
> I'm training one but the results are not as good so far. will let you know if I get it work. In the mean time, here are some super-cherry-picked golden samples:
> 
> ![arcane](https://user-images.githubusercontent.com/26464535/144731793-51f73a57-8f36-4243-b515-c83c4db5776c.jpeg)

Awesome! Is this a StyleGAN result then do a animegan training?

**Comment by onion-liu** [comment]
> Sure, have a look [here](https://we.tl/t-1wPGKRJlUJ)

The transfer has expired, can you share it again?

**Comment by Greg8978** [comment]
It's easy to reproduce, just collect arcane video(s) (I used youtube trailers), then 
`ffmpeg -i Arcanevideo.mp4  fps=0.5 arcane%d.jpg`

**Comment by bryandlee** [comment]
@bilal2vec These are super cherrypicked samples. The model's really fragile at the moment and have obvious normalization related artifacts for most of the images. I'm testing out some other techniques and trying to find a sweet spot between the quality and robustness.

@zhanglonghao1992  I didn't but it should help

@Sxela It's from a distilled pix2pix model

@rainsoulsrx @tinapan-pt  I've used about 500 images but it contains many duplicates cause it was taken from videos with limited characters

@chenhk-chn yup

**Comment by zhanglonghao1992** [comment]
@bryandlee Do you use pair data for training?

**Comment by rainsoulsrx** [comment]
> @bilal2vec These are super cherrypicked samples. The model's really fragile at the moment and have obvious normalization related artifacts for most of the images. I'm testing out some other techniques and trying to find a sweet spot between the quality and robustness.
> 
> @zhanglonghao1992 I didn't but it should help
> 
> @Sxela It's from a distilled pix2pix model
> 
> @rainsoulsrx @tinapan-pt I've used about 500 images but it contains many duplicates cause it was taken from videos with limited characters
> 
> @chenhk-chn yup

So besides these super cherrypicked samples, would  you please share some normal examples?

**Comment by bryandlee** [comment]
@zhanglonghao1992 yup

@rainsoulsrx  https://fragrant-chauffeur-53f.notion.site/Failures-d701a060e52046188b45823f56589093
It's pix2pixHD network (not the animegan architecture) with instance norm and I suspect that the black blobs are something similar to the "droplet artifact" in stylegan1.

**Comment by rainsoulsrx** [comment]
> @zhanglonghao1992 yup
> 
> @rainsoulsrx https://fragrant-chauffeur-53f.notion.site/Failures-d701a060e52046188b45823f56589093 It's pix2pixHD network with instance norm and I suspect that the black blobs are something similar to the "droplet artifact" in stylegan1.

Still not bad~
However in my experiment, I find it's hard to retain the 'style' when I train stylegan with less epoch, on the other hand, if I train stylegan more epoch, the 'style' become stronger, but I got more atifacts which will cause unpleasant data pair. just like below. So how do you make this balance?

![image](https://user-images.githubusercontent.com/24955512/145407925-8fca470d-4abb-4f23-8b99-0644b49f6352.png)

**Comment by Sxela** [comment]
> @zhanglonghao1992 yup
> 
> @rainsoulsrx https://fragrant-chauffeur-53f.notion.site/Failures-d701a060e52046188b45823f56589093 It's pix2pixHD network (not the animegan architecture) with instance norm and I suspect that the black blobs are something similar to the "droplet artifact" in stylegan1.

You can try batchnorm with spectral norm, seems to work okay for me

**Comment by bryandlee** [comment]
@Sxela Thanks for the suggestions. I've actually tried a number of architectures including BN/SN but didn't really get the quality that I wanted. +) I just found out that you've trained a nice model. If anyone interested in arcane style, check this out [Sxela/ArcaneGAN](https://github.com/Sxela/ArcaneGAN)

**Comment by Sxela** [comment]
> @Sxela Thanks for the suggestions. I've actually tried a number of architectures including BN/SN but didn't really get the quality that I wanted. +) I just found out that you've trained a nice model. If anyone interested in arcane style, check this out [Sxela/ArcaneGAN](https://github.com/Sxela/ArcaneGAN)

Thank you for mentioning, it means a lot to me! Can't wait to see your model from DeepStudio, it looks much better in my opinion, its temporal consistency is just outstanding.

**Comment by bryandlee** [comment]
@Sxela  No problem! Great work man

**Comment by seawater668** [comment]
> 练，如果我成功使用 ARCANE 图片，我会通知您

May I know how to train?

**Comment by seawater668** [comment]
> DeepStu

May I know how to train?

**Comment by watertianyi** [comment]
> @zhanglonghao1992是的
> 
> @rainsoulsrx https://fragrant-chauffeur-53f.notion.site/Failures-d701a060e52046188b45823f56589093 这是具有实例规范的 pix2pixHD 网络（不是 animegan 架构），我怀疑黑色斑点类似于 stylegan1 中的“液滴工件”。

You do not have access to Bryan's Notion. Please contact an admin to add you as a member.
